bpf/bpf_host.c

```diff
@@ -716,8 +921,12 @@ do_netdev(struct __ctx_buff *ctx, __u16 proto, bool from_host)
 		                              CTX_ACT_OK, METRIC_INGRESS);
 #endif
 	default:
+#ifdef ENABLE_HOST_FIREWALL
+		ret = DROP_UNKNOWN_L3;
```

Comment:
Should the default host firewall drop code be different?

Reply:
If I'm not mistaken, if `DROP_NOTIFY` is defined, the `EVENT_SOURCE` (i.e., `HOST_EP_ID` for the host endpoint) will be logged when the drop is logged. I don't know how often `DROP_NOTIFY` is set, but if that's considered to be the default, it might not be worth introducing new drop codes? On the other hand, the only downside of adding new drop codes is that it takes a couple new slots and the limit on the max. number of drop codes we can have (tied to metrics map's size) is still far.

Unrelated: Your comment made me realize that we're missing a call to `send_drop_notify_error()` here.

Reply:
> if DROP_NOTIFY is defined, the EVENT_SOURCE (i.e., HOST_EP_ID for the host endpoint) will be logged when the drop is logged.

My concern was if we end up reporting to the user "we're dropping this traffic because we don't support this protocol", but we really mean "the host firewall dropped the traffic and the firewall requires ipv4/ipv6". That said this is not likely to hit particularly often and it's likely unique at this `EVENT_SOURCE` anyway so we can figure out the reason here in retrospect. I agree that it's probably not necessary to make a distinction at this point.

FWIW, `DROP_NOTIFY `is typically defined to true. This is how we get notifications when packets are dropped.